internal/core/src/segcore/plan_c.cpp

```diff
@@ -76,91 +53,85 @@ ParsePlaceholderGroup(CSearchPlan c_plan,
                       const void* placeholder_group_blob,
                       const int64_t blob_size,
                       CPlaceholderGroup* res_placeholder_group) {
-    std::string blob_str((char*)placeholder_group_blob, blob_size);
-    auto plan = (milvus::query::Plan*)c_plan;
-
     try {
+        std::string blob_str(reinterpret_cast<char*>(const_cast<void*>(placeholder_group_blob)), blob_size);
+        auto plan = reinterpret_cast<milvus::query::Plan*>(c_plan);
         auto res = milvus::query::ParsePlaceholderGroup(plan, blob_str);
```

Comment:
Replace ParseFromString with ParseFromArray in milvuss::query::ParsePlaceholderGroup;